recipe_scrapers/rezeptwelt.py

```diff
@@ -31,15 +31,12 @@ def ingredients(self):
         return self.schema.ingredients()
 
     def instructions(self):
-        content = self.soup.find("ol", {"itemprop": "recipeInstructions"}).findAll(
-            "div", {"itemprop": "itemListElement"}
-        )
-        res = ""
-        for i in content:
-            steps = i.findAll("span", {"itemprop": "text"})
-            for step in steps:
-                res += normalize_string(step.text) + "\n"
-        return res
+        preparation_div = self.soup.find("div", id="preparationSteps")
+        instructions = preparation_div.find("span", itemprop="text").find_all("p")
+        instruction_texts = [instruction.get_text() for instruction in instructions]
+        normalized_texts = [normalize_string(text) for text in instruction_texts]
+        joined_instructions = "\n".join(filter(None, normalized_texts))
+        return joined_instructions
```

Comment:
```suggestion
        container = self.soup.find("div", id="preparationSteps").find("span", itemprop="text")
        instructions = [normalize_string(paragraph.text) for paragraph in container.find_all("p")]
        return "\n".join(filter(None, instructions))
```

Perhaps we can do more to make `normalize_string` -- or string filering/extraction in general -- neater in situations like this.  It's a fairly common pattern throughout the codebase (cc #617).